### ISSUE TITLE
missing FROM-clause entry when findAndCountAll on associations

### ISSUE THREAD

**Issue by maplesap** [sequelize/sequelize#3494]
I am trying to get back rows based on some field in an association: 

ie, get all items that are `green` and from `bob`:
```
models.item.findAndCountAll({ include: [ {model: model.child} ], where:
  Sequelize.and( { color: 'green'},  {'children.name': 'bob' })
});
```

Sequelize will throw an error:
```
{ name: 'SequelizeDatabaseError',
  message: 'missing FROM-clause entry for table "children"',
```

**Comment by mickhansen** [comment]
```js
models.item.findAndCountAll({
  where: {
    color: 'green'
  },
  include: [
    {model: model.child, where: {name: 'bob'}}
  ]
});
```
Or https://github.com/sequelize/sequelize/issues/3095

**Comment by sushantdhiman** [comment]
**SSCCE Test**

```js
/*
 * Copy this file to ./sscce.js
 * Add code from issue
 * npm run sscce-{dialect}
 */

var Sequelize = require('./index');
var sequelize = require('./test/support').createSequelizeInstance();

var M = sequelize.define('M', { color: Sequelize.STRING})
  , N = sequelize.define('N');

M.hasMany(N);

sequelize.sync({ force: true })
  .then(() => M.findAndCountAll({
    where: { color: 'green' },
    include: [{ model: N, where: { id: 1 }}]
  }))
  .then(console.log)
  .catch(console.error);
```

**Output**

```js
Using database:  sequelize-test-74
Executing (default): SELECT count("M"."id") AS "count" FROM "Ms" AS "M" INNER JOIN "Ns" AS "Ns" ON "M"."id" = "Ns"."MId" AND "Ns"."id" = 1 WHERE "M"."color" = 'green';
{ count: 0, rows: [] }
```

Fixed by #6190
